mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndex.java

```diff
@@ -1052,7 +1054,7 @@ private Query createTextQuery(SearchQuery.TextCriterion crit) throws Unsupported
             queryBuilder.add(createTermQuery(HEADERS_FIELD,value), BooleanClause.Occur.SHOULD);
             return queryBuilder.build();
         default:
-            throw new UnsupportedSearchException();
+            return new BooleanQuery.Builder().build(); // TODO: Implement
```

Comment:
I'm stuck here.
Because when input is `Scope.ATTACHMENTS`, it throw exception, and crash the JMAP flow.
Then I temporary return empty BooleanQuery.

Reply:
Rightnow, LuceneMessageSearchIndex doen't support ATTACHMENTS searching, 
while when jmap query filter.body="keyword", then it implicit include query ATTACHMENTS + BODY Text

Reply:
> Rightnow, LuceneMessageSearchIndex doen't support ATTACHMENTS searching

Maybe another ticket to add attachments index to Lucene and enable `jmap query filter.body="keyword", then it implicit include query ATTACHMENTS + BODY Text`?

Reply:
If the underlying search engine support attachment search then the JMAP layer position it as a search criteria otherwize no?

BTW maybe we should just contribute (optional) attachment indexing with Lucene?

Reply:
Split ticket for it https://github.com/linagora/james-project/issues/5306